compiler/semdata.nim

```diff
@@ -162,6 +162,7 @@ type
     lastTLineInfo*: TLineInfo
     sideEffects*: Table[int, seq[(TLineInfo, PSym)]] # symbol.id index
     inUncheckedAssignSection*: int
+    firstUsageOfForeignSym*: Table[int, tuple[ident: string, info: TLineInfo]] # (symbol.id, ...)
```

Comment:
Why do we need to care about the "first usage of a foreign sym"? Why not report misspellings regardless?

Reply:
I may be misunderstanding your question, but that was added for this:

> Checking the usage style of foreign symbols was completely disabled by
PR https://github.com/nim-lang/Nim/pull/19822. Now, all usage styles of a foreign symbol are compared against
the style of its first use in the module.

If we don't care about the style of a foreign symbol, that's fine, but we should be able to check for consistent style of that symbol in our module. But that means we must choose a certain style to compare all others to, and that seems like it should be the style of the first use of that foreign symbol in our module.

It was brought to my attention here: https://github.com/nim-lang/Nim/pull/19822#issuecomment-1221224687

Reply:
> but we should be able to check for consistent style of that symbol in our module. 

That's a new requirement that was not discussed anywhere previously afaik. And if it makes the implementation that more complex, I don't agree with the requirement.

Reply:
I removed it.